src/XPurchase/VirtoCommerce.XPurchase/CartAggregate.cs

```diff
@@ -697,14 +701,19 @@ public virtual Task<CartAggregate> UpdateVendor(LineItem lineItem, CartProduct c
             return Task.FromResult(this);
         }
 
-        public virtual Task<CartAggregate> UpdateOrganization(ShoppingCart cart, Member member)
+        public virtual async Task<CartAggregate> UpdateOrganization(ShoppingCart cart, Member member)
         {
             if (member is Contact contact && cart.Type != XPurchaseConstants.ListTypeName)
             {
                 cart.OrganizationId = contact.Organizations?.FirstOrDefault();
+                if (!string.IsNullOrEmpty(cart.OrganizationId))
+                {
+                    var org = await _memberService.GetByIdAsync(cart.OrganizationId);
```

Comment:
This code should not be inside `if (member is Contact contact && cart.Type != XPurchaseConstants.ListTypeName)`
Also, UpdateOrganization() call should not be iside `foreach (var lineItem in aggregate.LineItems)`